### PR TITLE
Load env variables based on `APP_ENV` or `-e` flag

### PR DESCRIPTION
**Pull request by beerose**
Closes: https://github.com/blitz-js/blitz/issues/2840

## Bug Checklist

- [ ] Integration test added (see [test docs](https://blitzjs.com/docs/contributing#running-tests) if needed)

## Feature Checklist

- [x] Integration test added (see [test docs](https://blitzjs.com/docs/contributing#running-tests) if needed)
- [x] Documentation added/updated (submit PR to [blitzjs.com repo `canary` branch](https://github.com/blitz-js/blitzjs.com/tree/canary)): https://github.com/blitz-js/blitzjs.com/pull/617

### CODE REVIEW

**Comment by beerose** [comment]
@flybayer I added the ability to set app env to `dev, start, build, export`. As for the other commands:
- `new` doesn't load anything from `.env` (or did I miss something)?
- `db` is deprectaed
- `prisma` — loads env variables inside of the `prisma` library, so that would need more changes right?
- `generate, codegen, install, routes` — doesn't need it?
- `console` — from what I see it doesn't load env variables, is that right?

**Comment by flybayer** [comment]
Ok this is looking good, but need a few more changes.

1. Set `APP_ENV` if `-e` present in [next cli entrypoint](https://github.com/blitz-js/blitz/blob/canary/nextjs/packages/next/bin/next.ts).
2. Replace every call to `dotenv-flow` with a call to the next-env package
3. Remove `dotenv-flow` dependency

Prisma loads `.env` but none of the other files. So we previously called `dotenv-flow` which loaded the other files. Now we are replacing that with next-env.

File: nextjs/packages/next-env/package.json [hunk 1]
```diff
@@ -1,7 +1,6 @@
 {
-  "private": true,
   "name": "@next/env",
-  "version": "11.1.0",
+  "version": "0.42.0",
```
Comment (beerose) [inline nextjs/packages/next-env/package.json:3]: update to latest

**Review by beerose — COMMENTED** [review]
* Add integration tests
* Test locally the next package
* Brandon has to do a release with new `@blitzjs/env` package
* `loadEnvConfig` should default to `process.cwd()`

File: nextjs/packages/next-env/index.ts [hunk 2]
```diff
@@ -78,8 +78,8 @@ export function loadEnvConfig(
   if (combinedEnv) return { combinedEnv, loadedEnvFiles: cachedLoadedEnvFiles }
 
   const isTest = process.env.NODE_ENV === 'test'
-  const mode = isTest ? 'test' : dev ? 'development' : 'production'
-  const dotenvFiles = [
+  let mode = isTest ? 'test' : dev ? 'development' : 'production'
```
Comment (beerose) [inline nextjs/packages/next-env/index.ts:81]:
```suggestion
  const mode = isTest ? 'test' : dev ? 'development' : 'production'
```

File: nextjs/test/integration/env-config-app-env/app/next.config.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
+module.exports = {
+  cleanDistDir: false,
+  target: 'experimental-serverless-trace',
```
Comment (flybayer) [inline nextjs/test/integration/env-config-app-env/app/next.config.js:3]: This line shouldn't be here

File: nextjs/test/integration/env-config-app-env/test/index.test.js [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+      '.env.staging.local',
+      '.env.production',
+    ].map((file) => join(appDir, file))
+
+    beforeAll(async () => {
+      nextConfigContent = await fs.readFile(nextConfigPath, 'utf8')
+      await fs.writeFile(
+        nextConfigPath,
+        nextConfigContent.replace(
+          '// update me',
```
Comment (flybayer) [inline nextjs/test/integration/env-config-app-env/test/index.test.js:128]: use this instead in config by default

**Review by flybayer — APPROVED** [review]

**Pull request merged**